main.cpp

```diff
@@ -126,384 +167,529 @@ struct CarWash
-    // Material (plastic, metal) (std::string)
     std::string material = "plastic";
-    // Caliber (float)
     float caliber = 5.56f;
-    // Current amount of bullets left (int)
     int currentCapacity = 30;
 
-    // Store rounds for pewpew
-    int holdRounds(int roundsLoaded, int roundsExpended); // Returns number of current rounds in capa
-    // Feed rounds into feeding system
+    int holdRounds(int roundsLoaded, int roundsExpended); 
     int feedRounds(bool isChamberJammed);
-    // Prevent loading of incorrect rounds
     void filterIncorrectRounds(bool isCaliberCorrect); 
 };
-/*
-Thing 8) Bullet
-5 properties:
-    1) Caliber (float)
-    2) Powder load (int)
-    3) Powder brand (std::string)
-    4) Bullet weight (grains) (float)
-    5) Bullet type (HP, AP, FMJ, HE) (std::string)
-3 things it can do:
-    1) Engage rifling in barrel
-    2) Accelerate when fired
-    3) Impact target downrange
- */
+
+int Magazine::holdRounds(int roundsLoaded, int roundsExpended)
+{
+    int roundsLeft = roundsLoaded - roundsExpended;
+    return roundsLeft;
+}
+
+int Magazine::feedRounds(bool isChamberJammed)
+{
+    int roundInChamber;
+    if (isChamberJammed == true)
+    {
+        roundInChamber = 0;
+    }
+    else
+    {
+        roundInChamber = 1;
+    }
+    return roundInChamber;
+}
+
+void Magazine::filterIncorrectRounds(bool isCaliberCorrect)
+{
+    if (isCaliberCorrect == false)
+    {
+        std::cout << "Incorrect caliber loaded! Unable to fire.\n";
+    }
+    return;
+}
+
+
 struct Bullet
 {
-    // Caliber (float)
     float caliber = 5.56f;
-    // Powder load (int)
     int powderLoad = 10;
-    // Powder brand (std::string)
     std::string powderBrand = "4198";
-    // Bullet weight (grains) (float)
     float weight = 10.0;
-    // Bullet type (HP, AP, FMJ, HE) (std::string)
     std::string type = "FMJ";
 
-    // Engage rifling in barrel
     void engageRifling(int threading, int acceleration);
-    // Accelerate when fired
-    int accelerateWhenFired(int powderLoad, float caliber, float barrelLength); // Returns maximum speed of bullet at end of barrel
-    // Impact target downrange
-    int impactTarget(int distance, int elevation, int speed, int mass);// Returns predicted speed of bullet when hitting target downrange
+    float accelerateWhenFired(int powderLoad, float caliber, float barrelLength); 
+    int impactTarget(int distance, int elevation, int speed, int mass);
 };
-/*
-Thing 9) Stock
-5 properties:
-    1) Is skeletonized (bool)
-    2) Is foldable (bool)
-    3) Has cheekrest (bool)
-    4) Has sling point (bool)
-    5) Weight (g) (float)
-3 things it can do:
-    1) Increase control of recoil in gun
-    2) Toggle folded/unfolded state
-    3) Provide support to firer's head when ADS
- */
+
+void Bullet::engageRifling(int threading, int acceleration)
+{
+    int riflingEngagement = threading * acceleration;
+    std::cout << riflingEngagement;
+    return;
+}
```

Comment:
Functions that don't return anything don't need a `return;` statement
- [ ] revise